indra/newview/lldrawpoolmaterials.cpp

```diff
@@ -247,21 +249,28 @@ void LLDrawPoolMaterials::renderDeferred(S32 pass)
         // upload matrix palette to shader
         if (rigged && params.mAvatar.notNull())
         {
-            if (params.mAvatar != lastAvatar)
+            if (params.mAvatar != lastAvatar || params.mSkinInfo != lastSkinInfo)
             {
+                llassert(params.mSkinInfo);
                 const LLVOAvatar::MatrixPaletteCache& mpc = params.mAvatar->updateSkinInfoMatrixPalette(params.mSkinInfo);
                 U32 count = static_cast<U32>(mpc.mMatrixPalette.size());
+                skipLastSkin = !bool(count);
+                lastAvatar = params.mAvatar;
```

Comment:
lastAvatar appeared to always be null, preventing this optimization from working